Documentation/guides/MSBuildReferenceAssemblies.md

```diff
@@ -0,0 +1,148 @@
+# MSBuild Reference Assemblies
+
+MSBuild and Roslyn support generating [reference
+assemblies][msbuild_refassemblies] by setting the
+`$(ProduceReferenceAssembly)` MSBuild property to True.
+
+Why do we care?  What's this mean?
+
+Assume you have a solution with two projects.  Project
+`Referenced.csproj` has no further references.  Project
+`Referencer.csproj` has a `@(ProjectReference)` to
+`Referenced.csproj`.
+
+The developer makes a change to something within `Referenced.csproj`.
+
+Question: Does `Referencer.csproj` need to be rebuilt?
+
+In the "original" MSBuild world -- the world that Xamarin.Android
+still lives in -- the answer is *yes*, `Referencer.csproj` *must
+**always*** be rebuilt, because the change to `Referenced.csproj`
+*may* contain an API breaking change which would prevent
+`Referencer.csproj` from building.
+
+In the new `$(ProduceReferenceAssembly)=True` world order, the answer
+is instead ***maybe***: `Referencer.csproj` *only* needs to be built
+if the reference assembly produced as part of the `Referenced.csproj`
+build is updated, which in turn only happens when the *public* API
+changes.  Meaning if a change *doesn't* alter the public API -- adding
+comments, fixing a method implementation, adding `private`/`internal`
+members, etc. -- then `Referencer.csproj` need not be rebuilt *at
+all*.
+
+In more concrete terms, assume you have a Xamarin.Forms solution
+containing a Xamarin.Forms PCL project and a referencing Android App
+project.  *Currently*, whenever the PCL project is changed, the App
+project must *always* be rebuilt.  In a
+`$(ProduceReferenceAssembly)=True` order, the App project would need
+to be rebuilt *less often*.
+
+[msbuild_refassemblies]: https://github.com/dotnet/roslyn/blob/master/docs/features/refout.md
+
```

Comment:
The above section was the relevant good stuff from here: https://github.com/xamarin/xamarin-android/wiki/Build-Performance-Ideas#embrace-reference-assemblies